### PR TITLE
style: don't assume HTML structure of rendered markdown in searchResultMatch

### PR DESCRIPTION
**Pull request by attfarhan**
Avoid relying on the structure of the response of our markdown renderer when rendering code blocks. Instead, we just want the text of any result wrapped in a code block, so we can pass it to the syntax highlighter.

Should mean that we avoid issues where syntax highlighting does not occur because we fail to detect a code block, such as https://github.com/sourcegraph/sourcegraph/issues/2325.

### CODE REVIEW

File: web/src/components/SearchResultMatch.tsx [hunk 1]
```diff
@@ -63,14 +63,11 @@ export class SearchResultMatch extends React.Component<SearchResultMatchProps, S
...
-                            // Get content between the outermost code tags.
-                            const codeContent = parser
-                                .parseFromString(markdownHTML, 'text/html')
-                                .querySelector('pre')!
-                                .innerHTML.toString()
+                            // Extract the text content of the result.
+                            const codeContent = parser.parseFromString(markdownHTML, 'text/html').body.innerText
                             if (codeContent) {
                                 return highlightCode({
                                     code: decode(codeContent),
```
Comment (felixfbecker) [inline web/src/components/SearchResultMatch.tsx]: `codeContent` is `innerText`, so it already doesn't contain HTML entities anymore - the `decode` would double-decode it

**Comment by slimsag** [comment]
@attfarhan Am I right that one outcome of this change would be that if we undo the recent changes to pkg/markdown, this change means this code will continue to work without reverting the commit that fixed https://github.com/sourcegraph/sourcegraph/issues/2325 again? If so, that is awesome. I've also filed https://github.com/sourcegraph/sourcegraph/issues/2369 as well to resolve my complaints against recent pkg/markdown changes which we discussed.

**Comment by slimsag** [comment]
@attfarhan I do echo what @felixfbecker said a bit here. Could we have some type of doc comment (not required for this PR) describing how this works? e.g.:

```
/*
The general process is:

1. We receive Markdown search results from the GraphQL search API.
2. We render that Markdown to HTML using the GraphQL markdown API.
3. We modify that HTML such that code blocks are highlighted using X because Y.
*/
```

Or some such, that way others can easily understand?

**Comment by attfarhan** [comment]
@slimsag that's correct. We don't care about the HTML structure that `pkg/markdown` or whatever markdown renderer we use spits out. 

@felixfbecker @slimsag here's a summary of what's happening: 

The `SearchResultMatch` component was built to render any arbitrary markdown search result, in anticipation of the search result providers extension API. So, we:
1. This component receives a single Markdown object (containing `html` and `text`) via props, representing the content of the search result. 
2. If the `html` field is an empty string (which will be the case for extensions), render the Markdown to HTML using the GraphQL Markdown API. If the html field is non-empty (which is the case for commit and diff searches, since we can render markdown on the server), just sanitize and use the HTML value.
3. If we know the Markdown result is a code block, then we extract the actual text content (the code itself), from the code block, and syntax highlight it using syntect via our GraphQL syntax highlighting API. For non-code blocks, just pass on the rendered markdown.

Why do we even render the markdown if we're gonna extract the text? The reason we do that is to handle non-code block markdown. I can probably rewrite this logic to make it clearer (e.g. have separate paths for code blocks/non-code blocks in the observable chain, so that we don't render markdown for code blocks). Also, since we have postponed the search result providers API, it might make sense to remove any logic that was written with that in mind altogether. I'll bring this up in the search kickoff meeting tomorrow.

Will also document this in the code in a separate PR.

**Pull request merged**